### PR TITLE
Remove environment from payment_method

### PR DESCRIPTION
**Pull request by jordan-brough**
Cherry-picked from https://github.com/spree/spree/pull/5882

See comments in https://github.com/solidusio/solidus/pull/229.

### CODE REVIEW

**Comment by magnusvk** [comment]
:+1:

**Comment by BenMorganIO** [comment]
yessssssss :+1: thanks @jordan-brough !!!

**Comment by gmacdougall** [comment]
Can you add something to the changelog about this?

**Comment by jordan-brough** [comment]
@gmacdougall like [this](https://github.com/solidusio/solidus/commit/cbc957bb53f0f1ced1fd0085905b7f4876191266)?

**Comment by gmacdougall** [comment]
Yup!

**Comment by jordan-brough** [comment]
I'll merge this after I have a chance to test it out on Bonobos code.

**Comment by jhawthorn** [comment]
:+1:

**Comment by jordan-brough** [comment]
Note to self: I was accidentally in this branch yesterday and noticed that Frontend in a Sandbox seemed to be displaying "New credit card" multiple times, I assume because it was finding PaymentMethods that it shouldn't have. I'm not sure if it's a sandbox setup issue or an issue w/ the migration or Frontend itself.  I will to look into that before merging this.

File: sample/db/samples/payment_methods.rb [hunk 1]
```diff
@@ -2,7 +2,6 @@
   {
     :name => "Credit Card",
     :description => "Bogus payment gateway for development.",
-    :environment => "development",
```
Comment (jhawthorn) [inline sample/db/samples/payment_methods.rb]: This is probably the issue you found @jordan-brough. Creating 4 credit card payment methods. We probably want to change the description also.

Comment (jordan-brough) [inline sample/db/samples/payment_methods.rb]: @jhawthorn awesome, thanks! I'll try to squash that early next week so we can get this merged in.

**Comment by cbrunsdon** [comment]
:+1:

**Comment by BenMorganIO** [comment]
I think this is the only PR left for making Spree 3 -> Solidus 1 a lot smoother.

**Comment by magnusvk** [comment]
Ok, an update here:
* Rebased ontop of latest master to address merge conflicts
* Removed duplicate payment methods in sample data; this fixes the sandbox issue @jordan-brough was seeing
* I tested this on the Bonobos store and ran into one issue: The solidus_multi_domain extension overrides `self.available` and was still trying to use `environment`. Curious what you think of https://github.com/jordan-brough/solidus/pull/1 to enable a more stable extension approach as seen in https://github.com/bonobos/solidus_multi_domain/pull/2. This feels perhaps more heavy-handed than needed; an alternative would be to just move over this logic from solidus_multi_domain to core.

Curious on thoughts on how to make this work with solidus_multi_domain and also another look-over of the updated PR.

**Comment by athal7** [comment]
Personally I would vote for moving the logic into Solidus since we are hoping to deprecate that extension anyways. If not, I like how you have implemented the ability to support the old method signature.

**Comment by jordan-brough** [comment]
> I would vote for moving the logic into Solidus

:100: 

I left some comments on https://github.com/jordan-brough/solidus/pull/1.  I think that's a great approach in general though I think in this specific case we can get away with something even simpler.

**Comment by magnusvk** [comment]
@solidusio/owners Rebased this _again_. As far as I can tell this is ready to merge. Cool to pull the trigger?

Also: New way of dealing with this change in https://github.com/solidusio/solidus_multi_domain/pull/13

**Comment by BenMorganIO** [comment]
Awesome. Can't wait to see this in master!

**Comment by gmacdougall** [comment]
:shipit:

**Comment by athal7** [comment]
:+1: on this change, interested to pull the multi-domain change into this codebase as well

**Comment by jordan-brough** [comment]
:+1: 

Left some comments on the associated multi-domain change.

> interested to pull the multi-domain change into this codebase as well

ditto

**Comment by jhawthorn** [comment]
:+1:

**Pull request merged**